sections/06_discussion.md

```diff
@@ -183,32 +188,83 @@ multitudinous layers and architectures available
-data. Unfortunately, it's precisely those data that would help to power deep
-learning in the domain. Though not a methodological consideration, efforts are
-underway to recognize those who promote an ecosystem of rigorous sharing and
-analysis [@doi:10.1038/ng.3830].
-
-*Reproducibiliy is important for science to progress. In the context of deep
-learning applied to advance human healthcare, does reproducibility have
-different requirements or alternative connotations? With vast hyperparameter
-spaces, massively heterogeneous and noisy biological data sets, and black box
-interpretability problems, how can we best ensure reproducible models? What
-might a clinician, or policy maker, need to see in a deep model in order to
-influence healthcare decisions? Or, is deep learning a hypothesis generation
-machine that requires manual validation? DeepChem and DragoNN are worth
-discussing here.*
+### Data, code, and model sharing
+
+A robust culture of data, code, and model sharing would do much to speed
+advances in this domain. The cultural barriers of data sharing in particular are
+perhaps best captured by the implications of using the term "research parasite"
+to describe scientists who use data from other researchers
+[@doi:10.1056/NEJMe1516564]. In short, a field that honors only discoveries and
+not the hard work of generating useful data will have difficulty encouraging
+scientists to share their hard-won data. Unfortunately, it's precisely those
+data that would help to power deep learning in the domain. Efforts are underway
+to recognize those who promote an ecosystem of rigorous sharing and analysis
+[@doi:10.1038/ng.3830].
+
+The sharing of high-quality, labeled datasets will be especially valuable.  In
+addition, researchers who invest time to preprocess datasets to be suitable for
+deep learning can make the preprocessing code (e.g., Basset
+[@tag:Kelley2016_basset]) and cleaned data (e.g., MoleculeNet
+[@tag:Wu2017_molecule_net]) publicly available to catalyze further research. As
+discussed in the Categorize section, there are complex privacy and legal issues
+involved in sharing patient data and deep learning models trained with it but
+also recent techniques to mitigate these concerns.  Furthermore, in some
+domains, some of the best training data has been generated privately, for
+example, high-throughput chemical screening data at pharmaceutical companies.
+There is little expectation or incentive for this private data to be shared.
+`TODO: this last line is fairly pessimistic, is there something better to say?
+Is it possible to create the right incentives and/or protect proprietary data?`
+
+Code sharing and open source licensing is essential for continued progress in
+this domain.  Even though some journals have improved requirements about code
+availability at the time of publication, there remain many opportunities to do
+better.  Preprints are firmly established in many areas of machine learning and
```

Comment:
Perhaps it is out of scope.  I brought it up because it has been a point of personal frustration.  More than once, there has been an exciting preprint on some biomedical deep learning topic that my group wanted to test out or build upon.  If the code isn't available, we're stuck.  Sometimes we don't even want to run the code, we just want to understand minute details of the training that aren't written down in the paper.  Without code, we're left guessing.

Part of what I wanted in this section is a call to go beyond the accepted practices in the community.  That includes sharing code when not required and highlighting example repositories that make the extra effort to provide pre-trained models, data pre-processing scripts, tutorials, etc.